src/crypto/crypto.ts

```diff
@@ -37,9 +74,16 @@ interface CredType {
   password: string;
 }
 
-const makeSecureBuffer = (password: string | undefined): SecureBuffer<string> => {
+type DecryptConfigV1 = {
+  tag: string;
+  iv: string;
+  secret: string;
+};
+type DecryptConfigV2 = DecryptConfigV1 & { tokens: string[] };
+
+const makeSecureBuffer = (password: string | undefined, encoding: 'utf8' | 'hex'): SecureBuffer<string> => {
   const newSb = new SecureBuffer<string>();
-  newSb.consume(Buffer.from(ensure(password), 'utf8'));
+  newSb.consume(Buffer.from(ensure(password), encoding));
```

Comment:
I guess it was because of the keychain code where password is `password?: string`:
https://github.com/forcedotcom/sfdx-core/blob/9e32558d895cae448e834e275f3ffbcaa59a1628/src/crypto/crypto.ts#L57-L69

and in the else block, `SecureBuffer.value` returns `Optional<string>`.

@shetzel `makeSecureBuffer` isn't exported so maybe we can make the change to enforce password is a string where `makeSecureBuffer` is called